towhee/engine/task_executor.py

```diff
@@ -26,51 +27,59 @@ class TaskExecutor(threading.Thread):
     Each device has one TaskExecutor.
     """
 
-    def __init__(self, name: str):
-        threading.Thread.__init__(self)
-        self.name = name
+    def __init__(self, name: str, cache_path: str = None):
+        super().__init__()
+        self._name = name
         self._task_queue = TaskQueue()
-        self._op_pool = OperatorPool()
-        self._need_stop = False
-
-    def acquire_op(self, name: str, init_args: dict) -> Operator:
-        """
-        Get an Operator from pool by name
+        self._op_pool = OperatorPool(cache_path=cache_path)
+        self._is_run = True
 
-        Args:
-            name: the Operator's unique name in hub
-            init_args: the Operator's initialization arguments
-        """
-        raise NotImplementedError
-
-    def release_op(self, op: Operator):
-        """
-        Release an Operator to pool
-        """
-        raise NotImplementedError
+    @property
+    def name(self):
+        return self._name
 
     # def set_op_parallelism(self, name: str, parallel: int = 1):
     #     """
     #     Set how many Operators with same name can be called concurrently.
     #     """
     #     raise NotImplementedError
 
-    def push_task(self, task: Task):
-        """
-        Push a task to task queue
+    def push_task(self, task: Task) -> bool:
+        """Push a task to the end of the task queue.
+
+        Args:
+            task:
+                Pre-initialized `Task` object to push onto the queue.
         """
-        # todo self._task_queue.push(task)
-        raise NotImplementedError
+        return self._task_queue.push(task)
 
     def run(self):
-        while not self._need_stop:
```

Comment:
One task_executor should have multi work thread to run tasks

Reply:
I'll add a TODO for this.